File: java/lasagna/src/main/java/Lasagna.java
public class Lasagna {
    
   public int expectedMinutesInOven(){
       return 40;
   }
    
    public int remainingMinutesInOven(int time){
        return 40 - time;
    }

    public int preparationTimeInMinutes(int number){
        return 2*number;
    }

    public int totalTimeInMinutes(int number, int time){
        return 40-this.remainingMinutesInOven(time)+ this.preparationTimeInMinutes(number);
    }
}

File: python/guidos-gorgeous-lasagna/lasagna.py
#time the lasagna should be in the oven according to the cookbook.
EXPECTED_BAKE_TIME = 40
PREPARATION_TIME = 2
def bake_time_remaining(elapsed_bake_time):
    '''
    '''
    return EXPECTED_BAKE_TIME - elapsed_bake_time
def preparation_time_in_minutes(number_of_layers):
    '''
    '''
    return number_of_layers * PREPARATION_TIME
def elapsed_time_in_minutes(number_of_layers, elapsed_bake_time):
    '''
    '''
    return preparation_time_in_minutes(number_of_layers) + elapsed_bake_time
